cli/azd/cmd/env.go

```diff
@@ -169,43 +170,57 @@ func envListCmd(rootOptions *commands.GlobalCommandOptions) *cobra.Command {
 }
 
 func envNewCmd(rootOptions *commands.GlobalCommandOptions) *cobra.Command {
-	actionFn := func(ctx context.Context, _ *cobra.Command, args []string, azdCtx *environment.AzdContext) error {
-		askOne := makeAskOne(rootOptions.NoPrompt)
-		azCli := commands.GetAzCliFromContext(ctx)
-
-		if err := ensureProject(azdCtx.ProjectPath()); err != nil {
-			return err
-		}
-
-		if err := tools.EnsureInstalled(ctx, azCli); err != nil {
-			return err
-		}
-
-		if len(args) == 1 {
-			rootOptions.EnvironmentName = args[0]
-		}
-
-		if _, err := createAndInitEnvironment(ctx, &rootOptions.EnvironmentName, azdCtx, askOne); err != nil {
-			return fmt.Errorf("creating new environment: %w", err)
-		}
-
-		if err := azdCtx.SetDefaultEnvironmentName(rootOptions.EnvironmentName); err != nil {
-			return fmt.Errorf("saving default environment: %w", err)
-		}
-
-		return nil
-	}
 	cmd := commands.Build(
-		commands.ActionFunc(actionFn),
+		&envNewAction{rootOptions: rootOptions},
 		rootOptions,
 		"new <environment>",
 		"Create a new environment.",
 		"",
 	)
-	cmd.Args = cobra.MaximumNArgs(1)
 	return cmd
 }
 
+type envNewAction struct {
+	rootOptions  *commands.GlobalCommandOptions
+	subscription string
+	location     string
+	principalID  string
+}
+
+func (en *envNewAction) SetupFlags(persis *pflag.FlagSet, local *pflag.FlagSet) {
+	local.StringVar(&en.subscription, "subscription", "", "Name or ID of an Azure subscription to use for the new environment")
+	local.StringVarP(&en.location, "location", "l", "", "Azure location for the new environment")
+	local.StringVar(&en.principalID, "principal", "", "Azure Active Directory principal name or ID to use for the new environment")
```

Comment:
Could we not include this parameter for now (and just keep the default where the CLI will compute the current principal and assign it as part of environment configuration)?  Folks who want to override can use `azd env set` to overwrite the value.

Giving this feedback because I don't love the strategy we have today for the principal ID and would like us to take fewer dependencies on it (and not have it be a first-class concept of the CLI).

Reply:
I understand your point. Are we fairly sure that in the process of obtaining the principal ID, us and az CLI will not prompt the user for additional information? Cause that would be difficult to handle by the IDE...

Reply:
>  Are we fairly sure that in the process of obtaining the principal ID, us and az CLI will not prompt the user for additional information

Yes - today it's completely non interactive, assuming you are logged in, we call an `az` command to try to query the graph API and if that fails fall back to trying to get the principal id from the token.

Reply:
OK, I have removed the principal ID parameter for now. We can always add it later as necessary.